nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateTableFetch.java

```diff
@@ -0,0 +1,254 @@
+        List<String> maxValueColumnNameList = StringUtils.isEmpty(maxValueColumnNames)
+                ? new ArrayList<>(0)
+                : Arrays.asList(maxValueColumnNames.split("\\s*,\\s*"));
+        List<String> maxValueClauses = new ArrayList<>(maxValueColumnNameList.size());
+
+        String columnsClause = null;
+        List<String> maxValueSelectColumns = new ArrayList<>(maxValueColumnNameList.size() + 1);
+        maxValueSelectColumns.add("COUNT(*)");
+
+        // For each maximum-value column, get a WHERE filter and a MAX(column) alias
+        maxValueColumnNameList.forEach(colName -> {
+            maxValueSelectColumns.add("MAX(" + colName + ") " + colName);
+            String maxValue = statePropertyMap.get(colName.toLowerCase());
+            if (!StringUtils.isEmpty(maxValue)) {
+                Integer type = columnTypeMap.get(colName.toLowerCase());
+                if (type == null) {
+                    // This shouldn't happen as we are populating columnTypeMap when the processor is scheduled.
+                    throw new IllegalArgumentException("No column type found for: " + colName);
+                }
+                // Add a condition for the WHERE clause
+                maxValueClauses.add(colName + " > " + getLiteralByType(type, maxValue, dbAdapter.getName()));
+            }
+        });
+
+        whereClause = StringUtils.join(maxValueClauses, " AND ");
+        columnsClause = StringUtils.join(maxValueSelectColumns, ", ");
+
+        // Build a SELECT query with maximum-value columns (if present)
+        final String selectQuery = dbAdapter.getSelectStatement(tableName, columnsClause, whereClause, null, null, null);
+
+        try (final Connection con = dbcpService.getConnection();
+             final Statement st = con.createStatement()) {
+
+            final Integer queryTimeout = context.getProperty(QUERY_TIMEOUT).asTimePeriod(TimeUnit.SECONDS).intValue();
+            st.setQueryTimeout(queryTimeout); // timeout in seconds
+
+            logger.debug("Executing {}", new Object[]{selectQuery});
+            ResultSet resultSet = st.executeQuery(selectQuery);
+            if (resultSet.next()) {
+                // Total row count is in the first column
+                int rowCount = resultSet.getInt(1);
+
+                // Update the state map with the newly-observed maximum values
+                ResultSetMetaData rsmd = resultSet.getMetaData();
+                for (int i = 2; i <= rsmd.getColumnCount(); i++) {
+                    String resultColumnName = rsmd.getColumnName(i).toLowerCase();
+                    int type = rsmd.getColumnType(i);
+                    try {
+                        String newMaxValue = getMaxValueFromRow(resultSet, i, type, statePropertyMap.get(resultColumnName.toLowerCase()), dbAdapter.getName());
+                        if (newMaxValue != null) {
+                            statePropertyMap.put(resultColumnName, newMaxValue);
```

Comment:
statePropertyMap is a mutable copy of the original state map, it's not applied to State until line 244 (after commit)